# Service to initialize localization

# Request
# Localization mode (e.g. OWN_SIDELINE)
int64 init_mode

# Some modes (e.g. POSITION) require additional information
float32 x # in meters
float32 y # in meters
float32 angle # in radians

# Constants
int64 OWN_SIDELINE=0  # Start at the sideline on the own half facing inwards (default game start)
int64 OPPONENT_HALF=1  # Start somewhere on the opponents side (needs a very large number of particles)
int64 OWN_HALF=2  # Start somewhere on our own side (needs a very large number of particles)
int64 POSITION=3  # Start at a specific position with unknown orientation
int64 POSE=4  # Start at a specific position and orientation

---
bool success # True if the localization was initialized successfully
